plugins/civchat2-paper/src/main/java/vg/civcraft/mc/civchat2/listeners/KillListener.java

```diff
@@ -1,94 +1,142 @@
+    ) {
+        this.config = Objects.requireNonNull(config);
+        this.dao = Objects.requireNonNull(dao);
+        this.settings = Objects.requireNonNull(settings);
     }
 
-    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
-    public void onPlayerKill(PlayerDeathEvent event) {
-        Player victim = event.getEntity();
-        if (victim.getKiller() == null) {
+    @EventHandler(
+        priority = EventPriority.MONITOR,
+        ignoreCancelled = true
+    )
+    public void handleDeathMessage(
+        final @NotNull PlayerDeathEvent event
+    ) {
+        switch (this.config.getDeathMessageType()) {
+            case DISABLED -> {
+                event.deathMessage(null);
+                return;
+            }
+            case VANILLA -> {
+                return;
+            }
+            case CUSTOM -> {
+                event.deathMessage(null);
+            }
+        }
+        final Player victim = event.getEntity();
+        if (!(victim.getKiller() instanceof final Player killer)) { // null check
             return;
         }
-        int killBroadcastRange = config.getKillBroadcastRange();
+        final int killBroadcastRange = this.config.getKillBroadcastRange();
         if (killBroadcastRange <= 0) {
             return;
         }
-        Player killer = victim.getKiller();
-        if (!settingsMan.getSendOwnKills(killer.getUniqueId())) {
+        if (!this.settings.getSendOwnKills(killer.getUniqueId())) {
             return;
         }
-        String msg;
-        ItemStack item = killer.getInventory().getItemInMainHand();
-        String victimFormattedName = String.format("%s%s", ChatColor.ITALIC, victim.getDisplayName());
-        String killerFormattedName = String.format("%s%s", ChatColor.ITALIC, killer.getDisplayName());
-
-        if (item == null || MaterialUtils.isAir(item.getType())) {
-            msg = String.format("%s%s %swas killed by %s %sby hand", ChatColor.DARK_GRAY, victimFormattedName,
-                ChatColor.DARK_GRAY, killerFormattedName, ChatColor.DARK_GRAY);
-        } else {
-            String itemName = ItemUtils.getItemName(item);
-            String displayName = ItemUtils.getDisplayName(item);
-            Boolean hasWordBank = Optional.ofNullable(ItemUtils.getItemMeta(item))
-                .map(r -> r.displayName())
-                .map(f -> f.children().size() > 0)
-                .orElse(false);
-            if (!hasWordBank) {
-                displayName = String.format("with a %s", itemName);
-            } else {
-                String killMessageFormat = settingsMan.getKillMessageFormat(killer.getUniqueId()).simpleDescription;
-                if (killMessageFormat.isBlank()) {
-                    displayName = String.format("%s%s", killMessageFormat, displayName);
-                } else {
-                    displayName = String.format("%s %s", killMessageFormat, displayName);
+        final Component killMessage; {
```

Comment:
huh???

Reply:
What's the problem?

Reply:
Since I cannot read your mind, I can only assume you found the scoped building of the kill message confusing, so I have moved it into its own function 👍